explore/src/main/scala/explore/attachments/ObsAttachmentsTable.scala

```diff
@@ -0,0 +1,407 @@
+              CheckedColumnId,
+              identity,
+              columnNames(CheckedColumnId),
+              cell =>
+                CheckboxView(
+                  id = NonEmptyString.unsafeFrom(s"checked-${cell.value.get.id}"),
+                  value = cell.value
+                    .withOnMod(oa =>
+                      ProgramQueries
+                        .updateObsAttachmentChecked[IO](props.pid, oa.id, oa.checked)
+                        .runAsync
+                    )
+                    .zoom(ObsAttachment.checked),
+                  label = ""
+                )
+            ).sortableBy(_.get.checked)
+          )
+      )
+      // Rows
+      .useMemoBy((props, _, _, _) => props.obsAttachments.reuseByValue)((_, _, _, _) =>
+        _.value.toListOfViews.sortBy(_.get.id)
+      )
+      .useReactTableBy((prop, ctx, _, cols, rows) =>
+        TableOptions(
+          cols,
+          rows,
+          getRowId = (row, _, _) => RowId(row.get.id.toString)
+        )
+      )
+      .useStateView(Enumerated[ObsAttachmentType].all.head)
+      .render { (props, ctx, action, _, _, table, newAttType) =>
+        import ctx.given
+
+        val dialogHeader = action.get match
+          case Action.None     => ""
+          case Action.Insert   => "Uploading Attachment"
+          case Action.Replace  => "Uploading Replacement"
+          case Action.Download => "Downloading Attachment"
+
+        def onInsertFileSelected(e: ReactEventFromInput): Callback =
+          val files = e.target.files.toList
+          (Callback(e.target.value = null) *>
+            action.set(Action.Insert) *>
+            insertAttachment(props, newAttType.get, files)
+              .guarantee(action.async.set(Action.None))
+              .runAsync)
+            .when_(files.nonEmpty)
+
+        val footer =
+          <.tr(
+            <.td(
+              ^.colSpan := 7,
+              <.div(
+                ExploreStyles.AttachmentsTableFooter,
+                EnumDropdownView(
+                  id = "attachment-type".refined,
+                  value = newAttType,
+                  clazz = ExploreStyles.FlatFormField
+                ),
+                <.label(
+                  labelButtonClasses,
+                  ^.htmlFor := "attachment-upload",
+                  Icons.FileArrowUp
+                ).withTooltip(
+                  tooltip = s"Upload new ${newAttType.get.shortName} attachment",
+                  placement = Placement.Right
+                ),
+                <.input(
+                  ExploreStyles.FileUpload,
+                  ^.tpe  := "file",
+                  ^.onChange ==> onInsertFileSelected,
+                  ^.id   := "attachment-upload",
+                  ^.name := "file"
+                )
+              )
+            )
+          )
+
+        <.div(
+          PrimeTable(
+            table,
+            striped = true,
+            compact = Compact.Very,
+            footerMod = footer,
```

Comment:
I just noticed we've been approaching other table footers in this same way. However, the `*mod` parameters of `PrimeTable` are designed so that you define the footers for columns and let the library do the rendering. You can add the  `^.colSpan := 7` via `footerCellMod`.

This obviously works, but just to bear in mind for the future: 
- This doesn't add the standard styling classes the facade inserts.
- I think a footer `tr` is supposed to have `th`s, not `td`s.

Reply:
Yeah. I just basically copied this from the brightnesses table. 😄 I'll look at this for the proposal attachments table and when I switch to virtualized table.